### PR TITLE
Don't hardcode fixed size for GetMemory in ChunkedTests

### PR DESCRIPTION
**Pull request by benaadams**
Pass the length from GetMemory to the receiver so it knows what chunk size to expect.

Should fix the ChunksWithGetMemoryWithInitialFlushWorks issue in the dep updates https://github.com/aspnet/AspNetCore/pull/7640

/cc @jkotalik @Tratcher

### CODE REVIEW

**Comment by Tratcher** [comment]
Superseded by https://github.com/aspnet/AspNetCore/pull/7671

File: src/Servers/Kestrel/test/InMemory.FunctionalTests/ChunkedResponseTests.cs [hunk 1]
```diff
@@ -517,6 +526,8 @@ public async Task ChunksWithGetMemoryLargeWriteBeforeFirstFlush()
         [Fact]
         public async Task ChunksWithGetMemoryWithInitialFlushWorks()
         {
+            var length = new IntAsRef();
```
Comment (pakrym) [inline src/Servers/Kestrel/test/InMemory.FunctionalTests/ChunkedResponseTests.cs]: @benaadams, why did you need this? Isn't local capturing enough?

Comment (benaadams) [reply src/Servers/Kestrel/test/InMemory.FunctionalTests/ChunkedResponseTests.cs]: Didn't trust it :) Was just being explicit that that was the intent
